tests/lib.rs

```diff
@@ -140,14 +140,20 @@ fn to_string_of_base_v0_error() {
 }
 
 #[test]
-fn explicit_v0_is_disallowed() {
+fn can_decode_binary_v0_cid() {
     use std::io::Cursor;
-    assert!(matches!(
-        Cid::read_bytes(Cursor::new([
-            0x00, 0x70, 0x12, 0x12, 0x12, 0x12, 0x12, 0x12, 0x12, 0x12, 0x12, 0x12
-        ])),
-        Err(Error::InvalidExplicitCidV0)
-    ));
+
+    let expected_cid = "QmZEeJY1YBBdQsyu7H93QXrnjjin55TBw7B3SGkp8fwVsx";
+    let cid = Cid::read_bytes(Cursor::new([
+        0, 1, 113, 18, 32, 161, 230, 156, 160, 24, 32, 72, 189, 105, 55, 21, 161, 41, 2, 236, 199,
```

Comment:
To me that looks like a binary multibase prefixed CID v1.

Background: In DAG-CBOR links (CIDs) are [prefixed with a null byte](https://github.com/ipld/ipld/blob/7ec959c351ff1b2e3f6b692601816dbc2f51b9e5/specs/codecs/dag-cbor/spec.md#links). This means that if you get your CID from DAG-CBOR, you need to strip off the first byte.

If you remove the first byte, and decode the CID you get: https://cid.ipfs.tech/#zdpuAwKS5MNGNpHh4CMj3UZYLLqWEyQR94GMZCFkNcLYbninn

It sounds like you're parsing DAG-CBOR manually, you might want to use an existing parser instead. I'd recommend https://github.com/ipld/serde_ipld_dagcbor.